homeassistant/components/tts/amazon_polly.py

```diff
@@ -0,0 +1,139 @@
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ["boto3==1.4.3"]
+
+CONF_REGION = "region_name"
+CONF_ACCESS_KEY_ID = "aws_access_key_id"
+CONF_SECRET_ACCESS_KEY = "aws_secret_access_key"
+CONF_PROFILE_NAME = "profile_name"
+ATTR_CREDENTIALS = "credentials"
+
+CONF_VOICE = "voice"
+CONF_OUTPUT_FORMAT = "output_format"
+CONF_SAMPLE_RATE = "sample_rate"
+CONF_TEXT_TYPE = "text_type"
+
+SUPPORTED_VOICES = ["Geraint", "Gwyneth", "Mads", "Naja", "Hans", "Marlene",
+                    "Nicole", "Russell", "Amy", "Brian", "Emma", "Raveena",
+                    "Ivy", "Joanna", "Joey", "Justin", "Kendra", "Kimberly",
+                    "Salli", "Conchita", "Enrique", "Miguel", "Penelope",
+                    "Chantal", "Celine", "Mathieu", "Dora", "Karl", "Carla",
+                    "Giorgio", "Mizuki", "Liv", "Lotte", "Ruben", "Ewa",
+                    "Jacek", "Jan", "Maja", "Ricardo", "Vitoria", "Cristiano",
+                    "Ines", "Carmen", "Maxim", "Tatyana", "Astrid", "Filiz"]
+
+SUPPORTED_OUTPUT_FORMATS = ["mp3", "ogg_vorbis", "pcm"]
+
+SUPPORTED_SAMPLE_RATES = ["8000", "16000", "22050"]
+
+SUPPORTED_SAMPLE_RATES_MAP = {
+    "mp3": ["8000", "16000", "22050"],
+    "ogg_vorbis": ["8000", "16000", "22050"],
+    "pcm": ["8000", "16000"]
+}
+
+SUPPORTED_TEXT_TYPES = ["text", "ssml"]
+
+CONTENT_TYPE_EXTENSIONS = {
+    "audio/mpeg": "mp3",
+    "audio/ogg": "ogg",
+    "audio/pcm": "pcm"
+}
+
+DEFAULT_VOICE = "Joanna"
+DEFAULT_OUTPUT_FORMAT = "mp3"
+DEFAULT_TEXT_TYPE = "text"
+
+DEFAULT_SAMPLE_RATES = {
+    "mp3": "22050",
+    "ogg_vorbis": "22050",
+    "pcm": "16000"
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_REGION, default="us-east-1"): cv.string,
+    vol.Inclusive(CONF_ACCESS_KEY_ID, ATTR_CREDENTIALS): cv.string,
+    vol.Inclusive(CONF_SECRET_ACCESS_KEY, ATTR_CREDENTIALS): cv.string,
+    vol.Exclusive(CONF_PROFILE_NAME, ATTR_CREDENTIALS): cv.string,
+    vol.Optional(CONF_VOICE, default=DEFAULT_VOICE):
+        vol.In(SUPPORTED_VOICES),
+    vol.Optional(CONF_OUTPUT_FORMAT, default=DEFAULT_OUTPUT_FORMAT):
+        vol.In(SUPPORTED_OUTPUT_FORMATS),
+    vol.Optional(CONF_SAMPLE_RATE): vol.All(cv.string,
+                                            vol.In(SUPPORTED_SAMPLE_RATES)),
+    vol.Optional(CONF_TEXT_TYPE, default=DEFAULT_TEXT_TYPE):
+        vol.In(SUPPORTED_TEXT_TYPES),
+})
+
+
+def get_engine(hass, config):
+    """Setup Amazon Polly speech component."""
+    # pylint: disable=import-error
+    output_format = config.get(CONF_OUTPUT_FORMAT)
+    if config.get(CONF_SAMPLE_RATE) is None:
+        config[CONF_SAMPLE_RATE] = DEFAULT_SAMPLE_RATES[output_format]
+    sample_rate = config.get(CONF_SAMPLE_RATE)
+    if sample_rate not in SUPPORTED_SAMPLE_RATES_MAP.get(output_format):
+        fmt_str = "{} is not a valid sample rate for {}"
+        _LOGGER.error(fmt_str.format(sample_rate, output_format))
+        return None
+
+    import boto3
```

Comment:
@pvizeli It should yes but I couldn't find a way to ensure that the given sample rate is valid for the given output format all in Voluptuous, so that check is done first in `get_engine`.